tests/acceptance/builds/prioritize-test.js

```diff
@@ -0,0 +1,84 @@
+import { module, test } from 'qunit';
+import { visit, click, triggerKeyEvent } from '@ember/test-helpers';
+import { setupApplicationTest } from 'travis/tests/helpers/setup-application-test';
+import signInUser from 'travis/tests/helpers/sign-in-user';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+
+module('Acceptance | builds/prioritize', function (hooks) {
+  setupApplicationTest(hooks);
+  setupMirage(hooks);
+
+  hooks.beforeEach(function () {
+    const currentUser = this.server.create('user');
+    signInUser(currentUser);
+  });
+
+  test('prioritizing build button does not exsist if the build is not having permission to prioritize becuase of the organization', async function (assert) {
+    let repository =  this.server.create('repository', { slug: 'travis-ci/travis-web' });
+    let branch = this.server.create('branch', { repository, name: 'acceptance-tests', default_branch: true });
+    let gitUser = this.server.create('git-user', { name: 'Mr T' });
+    let commit = this.server.create('commit', { author: gitUser, committer: gitUser, committer_name: 'Mr T', message: 'This is a message' });
+    let build = this.server.create('build', { number: '5', state: 'queued', permissions: { 'prioritize': false }, repository, commit, branch, priority: true, });
+    let job = this.server.create('job', { number: '1234.1', state: 'queued', repository, commit, build });
+    this.server.create('log', { id: job.id });
+
+    await visit('/travis-ci/travis-web');
+    assert.dom('[data-test-repo-actions-prioritize-button]').doesNotExist();
+  });
+
+  test('prioritizing build button does not exsist if the build is in the finished states', async function (assert) {
+    let repository =  this.server.create('repository', { slug: 'travis-ci/travis-web' });
+    let branch = this.server.create('branch', { repository, name: 'acceptance-tests', default_branch: true });
+    let gitUser = this.server.create('git-user', { name: 'Mr T' });
+    let commit = this.server.create('commit', { author: gitUser, committer: gitUser, committer_name: 'Mr T', message: 'This is a message' });
+    let build = this.server.create('build', { number: '5', state: 'passed', permissions: { 'prioritize': false }, repository, commit, branch, priority: true, });
```

Comment:
The `permissions.prioritize: false` and `priority: true` will both prevent the button from showing up on the page, right? Therefore, the `build.state` doesn't actually play any role here.

Reply:
Allowing the build to be prioritized via permissions + making it not yet prioritized, so that it _could_ be prioritized:
```suggestion
    let build = this.server.create('build', { number: '5', state: 'passed', permissions: { 'prioritize': true }, repository, commit, branch, priority: false, });
```